dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlVisitor.java

```diff
@@ -123,8 +107,25 @@ public void visit(Function function) {
 
     @Override
     public void visit(SignedExpression signedExpression) {
-        // TODO This translation should be implemented
-        throw new RuntimeException("Extraction of condition not yet implemented");
+        signedExpression.getExpression().accept(this);
+        SqlCondition sqlCondition = stack.pop();
+        if (sqlCondition instanceof SqlLiteralValue) {
+            SqlLiteralValue sqlLiteralValue = (SqlLiteralValue) sqlCondition;
+            SqlLiteralValue negated;
+            if (sqlLiteralValue instanceof SqlBigIntegerLiteralValue) {
+                SqlBigIntegerLiteralValue sqlBigIntegerLiteralValue = (SqlBigIntegerLiteralValue) sqlLiteralValue;
+                negated = new SqlBigIntegerLiteralValue(sqlBigIntegerLiteralValue.getBigInteger().negate());
+            } else if (sqlLiteralValue instanceof SqlBigDecimalLiteralValue) {
+                SqlBigDecimalLiteralValue sqlBigDecimalLiteralValue = (SqlBigDecimalLiteralValue) sqlLiteralValue;
+                negated = new SqlBigDecimalLiteralValue(sqlBigDecimalLiteralValue.getBigDecimal().negate());
+
+            } else {
+                throw new RuntimeException("Extraction of condition not yet implemented");
```

Comment:
could be useful to add `sqlLiteralValue .getClass` to the error message

Reply:
thanks